Mage/src/main/java/mage/game/mulligan/SmoothedLondonMulligan.java

```diff
@@ -0,0 +1,59 @@
+package mage.game.mulligan;
+
+import mage.cards.Card;
+import mage.cards.CardsImpl;
+import mage.constants.CardType;
+import mage.game.Game;
+import mage.players.Player;
+import mage.util.RandomUtil;
+
+import java.util.*;
+
+public class SmoothedLondonMulligan extends LondonMulligan {
+
+    public SmoothedLondonMulligan(int freeMulligans) {
+        super(freeMulligans);
+    }
+
+    SmoothedLondonMulligan(final SmoothedLondonMulligan mulligan) {
+        super(mulligan);
+    }
+
+    @Override
+    public void drawHand(int numCards, Player player, Game game){
+        List<Card> library = player.getLibrary().getCards(game);
+        if (library.size() >= numCards*2 && numCards > 1) {
+            double land_ratio = (double) library.stream().filter(w -> w.getCardType().contains(CardType.LAND)).count() / (double) library.size();
```

Comment:
You probably want to also count the mdfc that are lands only on the back? Not sure.

Reply:
Good catch, thanks! I'll count them as half a land in the library and randomly as a land or spell in the hand.
This randomness is because counting it as a half land in the hand would mean that you can adjust the deck's land ratio to be (integer vs X.5)/7 and through that significantly change the odds of drawing an MDFC.